src/gc/gc.cpp

```diff
@@ -16172,6 +16169,28 @@ void gc_heap::gc1()
             fire_pevents();
             pm_full_gc_init_or_clear();
 
+            // distribute gen0_must_clear_bricks to all heaps -
+            // if one heap encountered an interior pointer during this GC,
+            // the next GC might see one on another heap
+
+            // first, compute max
+            int max_gen0_must_clear_bricks = 0;
+            for (int i = 0; i < gc_heap::n_heaps; i++)
```

Comment:
>for (int i = 0; i < gc_heap::n_heaps; i++) [](start = 12, length = 42)

could we do this in the for loop above with 
hp->decommit_ephemeral_segment_pages();
since we already had to loop through all heaps? and then if max_gen0_must_clear_bricks is > 0 we can do another for loop to update

Reply:
This would work, because the data should be stable at this point - after all, it's computed during mark_phase. Most difficult problem is where to put the comments… Done.